endpoints/openrtb2/amp_auction.go

```diff
@@ -138,6 +139,12 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 	activityControl := privacy.ActivityControl{}
 
 	defer func() {
+		// if AmpObject.AuctionResponse is nil then collect nonbids from all stage outcomes and set it in the AmpObject.SeatNonBid
+		// Nil AmpObject.AuctionResponse indicates the occurrence of a fatal error.
+		if ao.AuctionResponse == nil {
```

Comment:
One direct way would be to call below 2 lines after every `return` statement.
`seatNonBid.Append(getNonBidsFromStageOutcomes(hookExecutor.GetOutcomes()))`
`ao.SeatNonBid = seatNonBid.Get()`

But currently there are around 6/7 `return` statement in `Auction()` & `AmpAuction()` function. To avoid the repetitive code, added this block in defer statement.
Thoughts ??

Reply:
You could perhaps have a `foundErrors` boolean flag that starts false, and then is set to true whenever errors are detected. Then use this flag in the defer rather than checking for nil, resulting in a more straightforward way to satisfy the conditional, and futureproofing against any possible change where either an error happens after `ao.AuctionResponse` is set or it stays nil despite no error.

Reply:
Used `foundError` instead of `ao.Response` nil check.